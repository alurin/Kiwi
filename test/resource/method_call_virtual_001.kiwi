#!/usr/bin/env kiwi
/* Test script for call method from child class */

class A {
    string getClassName() {
        return "A";
    }

    void test() {
        print "Call "; print getClassName(); print "::test()\n";
    }
}

class B {
    inherit A;

    string getClassName() {
        return "B";
    }
}

class C {
    inherit A;

    string getClassName() {
        return "C";
    }
}

class D {
    inherit A;

    string getClassName() {
        return "D";
    }
}

void main() {
    auto $a = new A();
    auto $b = new B();
    auto $c = new C();
    auto $d = new D();

    $a.test();
    $b.test();
    $c.test();
    $d.test();
}
