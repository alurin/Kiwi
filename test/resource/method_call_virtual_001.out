Call A::test()
Call B::test()
Call C::test()
Call D::test()
